aiosmtpd/smtp.py

```diff
@@ -535,14 +543,9 @@ def smtp_DATA(self, arg):
         received_data = EMPTYBYTES.join(data)
         if self._decode_data:
             received_data = received_data.decode('utf-8')
-        args = (self.peer, self.mailfrom, self.rcpttos, received_data)
-        kwargs = {}
-        if not self._decode_data:
-            kwargs = {
-                'mail_options': self.mail_options,
-                'rcpt_options': self.rcpt_options,
-                }
-        kwargs.update({'loop': self.loop})
+        self.message.received_data = received_data
+        args = (self.session, self.message)
+        kwargs = {'loop': self.loop}
```

Comment:
Hmm.  Could you elaborate please?  I looked briefly at #41 but need to spend a bit more time on that.

But this is really a question about the API for `process_message()`.  I'd like to get rid of the `**kws` argument to `process_message()` so that to me means either having an explicit `loop` argument, or moving `loop` to the session.  It's a bit of an abstraction breakage to put it on the session though.  E.g. what if one day we ported this to a different library (e.g. [curio](http://curio.readthedocs.io/en/latest/) or [trio](http://trio.readthedocs.io/en/latest/))?   OTOH, that'll probably never happen. :)

OTOH, I think in #41 you get rid of the synchronous `process_message()` calls (which is fine with me) so I'm happy to defer this decision until #41.